hexa/pipelines/middlewares.py

```diff
@@ -0,0 +1,36 @@
+import binascii
+from logging import getLogger
+
+from django.core.signing import BadSignature, Signer
+from django.http import HttpRequest
+
+from hexa.pipelines.models import PipelineRun
+
+from .authentication import PipelineRunUser
+
+logger = getLogger(__name__)
+
+
+def pipeline_run_authentication_middleware(get_response):
+    """This middleware allows an Pipeline v2 run to authenticate through a simple token"""
+
+    def middleware(request: HttpRequest):
+        try:
+            auth_type, token = request.headers["Authorization"].split(" ")
+            if auth_type.lower() == "bearer":
+                token = Signer().unsign_object(token)
+                pipeline_run = PipelineRun.objects.get(webhook_token=token)
```

Comment:
Is `webhook_token` still appropriate?

Reply:
Will update that later when permission archi is clearer